src/main.rs

```diff
@@ -163,14 +163,14 @@ impl PrirodaSender {
         T: rocket::response::Responder<'r> + Send + 'static,
         F: FnOnce(&mut PrirodaContext<'_, '_>) -> T + Send + 'static,
     {
-        let (future, promise) = future_promise();
+        let (tx, rx) = mpsc::sync_channel(0);
         let sender = self.0.lock().unwrap_or_else(|err| err.into_inner());
         match sender.send(Box::new(move |pcx: &mut PrirodaContext<'_, '_>| {
-            promise.set(f(pcx));
+            tx.send(f(pcx)).unwrap();
         })) {
-            Ok(()) => match future.value() {
```

Comment:
Certainly for reviewing purposes, this whole function is a bit messy in terms of control flow. I do believe that these changes are equivalent.

Reply:
Opened #47.